### PR TITLE
Document System.IO.FileSystemAclExtensions

### PR DESCRIPTION
**Pull request by carlossanlop**
Jeremy and I own these.
I wrote the Create methods.

Source: https://source.dot.net/#System.IO.FileSystem.AccessControl/System/IO/FileSystemAclExtensions.cs,28

### CODE REVIEW

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 1]
```diff
@@ -110,9 +142,9 @@
         <Parameter Name="directoryInfo" Type="System.IO.DirectoryInfo" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="directoryInfo">A directory information object that wraps the existing directory from which to obtain the security information.</param>
```
Comment (JeremyKuhne) [inline xml/System.IO/FileSystemAclExtensions.xml]: Seems a little too wordy. `The existing directory from which to obtain the security information.` ?

Comment (JeremyKuhne) [reply xml/System.IO/FileSystemAclExtensions.xml]: Same comment applies to the other params. Whatever we consider standard is fine.

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 2]
```diff
@@ -76,16 +88,36 @@
...
+        <exception cref="ArgumentNullException">
+          <paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        <exception cref="ArgumentOutOfRangeException">
+          <paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+
+-or-
+
+<paramref name="bufferSize" /> is not a positive number.</exception>
+        <exception cref="DirectoryNotFoundException">Could not find a part of the path.</exception>
+        <exception cref="IOException">An I/O error occurs.</exception>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="IOException">An I/O error occurred.</exception>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 3]
```diff
@@ -138,9 +170,9 @@
         <Parameter Name="fileInfo" Type="System.IO.FileInfo" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="fileInfo">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileInfo">A file information object that wraps the file from which to obtain the security information.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileInfo">The file from which to obtain the security information.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 4]
```diff
@@ -166,10 +198,13 @@
         <Parameter Name="fileStream" Type="System.IO.FileStream" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="fileStream">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileStream">A file stream that wraps an existing file from which to obtain the security information.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileStream">An existing file from which to obtain the security information.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 5]
```diff
@@ -195,10 +230,10 @@
         <Parameter Name="includeSections" Type="System.Security.AccessControl.AccessControlSections" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <param name="includeSections">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="directoryInfo">A directory information object that wraps an existing directory from which to obtain the security information.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="directoryInfo">An existing directory from which to obtain the security information.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 6]
```diff
@@ -225,10 +260,10 @@
         <Parameter Name="includeSections" Type="System.Security.AccessControl.AccessControlSections" />
       </Parameters>
       <Docs>
-        <param name="fileInfo">To be added.</param>
-        <param name="includeSections">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileInfo">A file information object that wraps an existing file from which to obtain the security information.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileInfo">An existing file from which to obtain the security information.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 7]
```diff
@@ -255,10 +290,12 @@
         <Parameter Name="directorySecurity" Type="System.Security.AccessControl.DirectorySecurity" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <param name="directorySecurity">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="directoryInfo">A directory information object that wraps an existing directory.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="directoryInfo">An existing directory.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 8]
```diff
@@ -255,10 +290,12 @@
...
+        <param name="directorySecurity">An object that contains the security information to apply to the directory.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="directorySecurity">The security information to apply to the directory.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 9]
```diff
@@ -284,10 +321,12 @@
         <Parameter Name="fileSecurity" Type="System.Security.AccessControl.FileSecurity" />
       </Parameters>
       <Docs>
-        <param name="fileInfo">To be added.</param>
-        <param name="fileSecurity">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="fileInfo">A file information object that wraps an existing file.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileInfo">An existing file.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 10]
```diff
@@ -284,10 +321,12 @@
...
+        <param name="fileSecurity">An object that contains the security information to apply to the file.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileSecurity">The security information to apply to the file.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 11]
```diff
@@ -313,10 +352,13 @@
         <Parameter Name="fileSecurity" Type="System.Security.AccessControl.FileSecurity" />
       </Parameters>
       <Docs>
-        <param name="fileStream">To be added.</param>
-        <param name="fileSecurity">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="fileStream">A file stream object that wraps an existing file.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileStream">An existing file.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 12]
```diff
@@ -313,10 +352,13 @@
...
+        <param name="fileSecurity">An object that contains the security information to apply to the file.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileSecurity">The security information to apply to the file.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 13]
```diff
@@ -225,10 +260,10 @@
...
       </Parameters>
       <Docs>
-        <param name="fileInfo">To be added.</param>
-        <param name="includeSections">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileInfo">A file information object that wraps an existing file from which to obtain the security information.</param>
+        <param name="includeSections">The desired access control sections to retrieve from the file.</param>
+        <summary>Returns the security information of a file.</summary>
+        <returns>An object that contains the security descriptors of the specified access control sections of the file.</returns>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <returns>The security descriptors of the specified access control sections of the file.</returns>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 14]
```diff
@@ -195,10 +230,10 @@
...
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <param name="includeSections">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="directoryInfo">A directory information object that wraps an existing directory from which to obtain the security information.</param>
+        <param name="includeSections">The desired access control sections to retrieve.</param>
+        <summary>Returns the security information of a directory.</summary>
+        <returns>An object that contains the security descriptors of the specified access control sections of the directory.</returns>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <returns>The security descriptors of the specified access control sections of the directory.</returns>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 15]
```diff
@@ -166,10 +198,13 @@
         <Parameter Name="fileStream" Type="System.IO.FileStream" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="fileStream">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileStream">A file stream that wraps an existing file from which to obtain the security information.</param>
+        <summary>Returns the security information of a file.</summary>
+        <returns>An object that contains the security descriptors of all the access control sections of the file.</returns>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <returns>The security descriptors of all the access control sections of the file.</returns>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 16]
```diff
@@ -138,9 +170,9 @@
         <Parameter Name="fileInfo" Type="System.IO.FileInfo" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="fileInfo">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="fileInfo">A file information object that wraps the file from which to obtain the security information.</param>
+        <summary>Returns the security information of a file.</summary>
+        <returns>An object that contains the security descriptors of all the access control sections of the file.</returns>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <returns>The security descriptors of all the access control sections of the file.</returns>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 17]
```diff
@@ -110,9 +142,9 @@
         <Parameter Name="directoryInfo" Type="System.IO.DirectoryInfo" RefType="this" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="directoryInfo">A directory information object that wraps the existing directory from which to obtain the security information.</param>
+        <summary>Returns the security information of a directory.</summary>
+        <returns>An object that contains the security descriptors of all the access control sections of the directory.</returns>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <returns>The security descriptors of all the access control sections of the directory.</returns>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 18]
```diff
@@ -76,16 +88,36 @@
...
-        <param name="mode">To be added.</param>
-        <param name="rights">To be added.</param>
-        <param name="share">To be added.</param>
-        <param name="bufferSize">To be added.</param>
-        <param name="options">To be added.</param>
-        <param name="fileSecurity">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="fileInfo">A file information object that represents a file that does not exist yet.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="fileInfo">A file that does not exist yet that will be created by the method.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 19]
```diff
@@ -46,10 +46,22 @@
         <Parameter Name="directorySecurity" Type="System.Security.AccessControl.DirectorySecurity" Index="1" FrameworkAlternate="dotnet-plat-ext-3.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <param name="directorySecurity">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="directoryInfo">A directory information object that represents a directory that does not exist yet.</param>
+        <param name="directorySecurity">An object that determines the access control and audit security for the directory.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="directorySecurity">The access control and audit security for the directory.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 20]
```diff
@@ -46,10 +46,22 @@
         <Parameter Name="directorySecurity" Type="System.Security.AccessControl.DirectorySecurity" Index="1" FrameworkAlternate="dotnet-plat-ext-3.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="directoryInfo">To be added.</param>
-        <param name="directorySecurity">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="directoryInfo">A directory information object that represents a directory that does not exist yet.</param>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <param name="directoryInfo">A directory that does not exist yet that will be created by the method.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 21]
```diff
@@ -23,7 +23,7 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides Windows-specific static extension methods that allow manipulating Access Control List (ACL) security attributes for files and directories.</summary>
```
Comment (gewarren) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
    <summary>Provides Windows-specific static extension methods for manipulating Access Control List (ACL) security attributes for files and directories.</summary>
```

File: xml/System.IO/FileSystemAclExtensions.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.IO/FileSystemAclExtensions.xml]: Sure, will change it:
```suggestion
        <param name="directoryInfo">The existing directory from which to obtain the security information.</param>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 22]
```diff
@@ -46,10 +46,22 @@
...
+
+## Remarks
+
+This extension method was added to .NET Core to bring the functionality that was provided by the `System.IO.DirectoryInfo.Create(System.Security.AccessControl.DirectorySecurity)` .NET Framework method.
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.ArgumentNullException">
+          <paramref name="directoryInfo" /> or <paramref name="directorySecurity" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.DirectoryNotFoundException">Could not find a part of the path.</exception>
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.IO.DirectoryNotFoundException">Could not find a part of the path.</exception>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 23]
```diff
@@ -76,16 +88,36 @@
...
+        <exception cref="ArgumentException">The <paramref name="rights" /> and <paramref name="mode" /> combination is invalid.</exception>
+        <exception cref="ArgumentNullException">
+          <paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        <exception cref="ArgumentOutOfRangeException">
+          <paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+
+-or-
+
+<paramref name="bufferSize" /> is not a positive number.</exception>
+        <exception cref="DirectoryNotFoundException">Could not find a part of the path.</exception>
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.IO.DirectoryNotFoundException">Could not find a part of the path.</exception>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 24]
```diff
@@ -76,16 +88,36 @@
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This extension method was added to .NET Core to bring the functionality that was provided by the `System.IO.FileStream.#ctor(System.String,System.IO.FileMode,System.Security.AccessControl.FileSystemRights,System.IO.FileShare,System.Int32,System.IO.FileOptions,System.Security.AccessControl.FileSecurity)` .NET Framework constructor.
+
+          ]]></format>
+        </remarks>
+        <exception cref="ArgumentException">The <paramref name="rights" /> and <paramref name="mode" /> combination is invalid.</exception>
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.ArgumentException">The <paramref name="rights" /> and <paramref name="mode" /> combination is invalid.</exception>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 25]
```diff
@@ -76,16 +88,36 @@
...
+        <exception cref="ArgumentNullException">
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException">
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 26]
```diff
@@ -76,16 +88,36 @@
...
+          <paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        <exception cref="ArgumentOutOfRangeException">
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 27]
```diff
@@ -76,16 +88,36 @@
...
+          <paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+
+-or-
+
+<paramref name="bufferSize" /> is not a positive number.</exception>
+        <exception cref="DirectoryNotFoundException">Could not find a part of the path.</exception>
+        <exception cref="IOException">An I/O error occurred.</exception>
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.IO.IOException">An I/O error occurred.</exception>
```

File: xml/System.IO/FileSystemAclExtensions.xml [hunk 28]
```diff
@@ -76,16 +88,36 @@
...
+        <exception cref="UnauthorizedAccessException">Access to the path is denied.</exception>
```
Comment (carlossanlop) [inline xml/System.IO/FileSystemAclExtensions.xml]:
```suggestion
        <exception cref="T:System.UnauthorizedAccessException">Access to the path is denied.</exception>
```
